work/v1/types.go

```diff
@@ -218,11 +280,68 @@ type ManifestCondition struct {
 	// +required
 	ResourceMeta ManifestResourceMeta `json:"resourceMeta"`
 
+	// StatusFeedback represents the values of the feild synced back defined in statusFeedbacks
+	// +optional
+	StatusFeedbacks StatusFeedbackResult `json:"statusFeedback,omitempty"`
+
 	// Conditions represents the conditions of this resource on a managed cluster.
 	// +required
 	Conditions []metav1.Condition `json:"conditions"`
 }
 
+// StatusFeedbackResult represents the values of the feild synced back defined in statusFeedbacks
+type StatusFeedbackResult struct {
+	// Values represents the synced value of the interested field.
+	// +listType:=map
+	// +listMapKey:=name
+	// +optional
+	Values []FeedbackValue `json:"values,omitempty"`
+}
+
+type FeedbackValue struct {
+	// name represents the aliase name for this field. It is the same as what is specified
+	// in spec.StatuFeedbacks.
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+
+	// Value is the value of the status field.
+	// The value of the status field can only be integer, string or boolean.
+	// +kubebuilder:validation:Required
+	// +required
+	Value FieldValue `json:"fieldValue"`
+}
+
+// FieldValue is the value of the status field.
+// The value of the status field can only be integer, string or boolean.
+type FieldValue struct {
+	// Type represents the type of the value, it can be integer, string or boolean.
+	// +kubebuilder:validation:Required
+	// +required
+	Type ValueType `json:"type"`
+
+	// Integer is the integer value when type is integer.
+	// +optional
+	Integer int32 `json:"integer,omitempty"`
```

Comment:
how you think if make these values pointer? so we can leave the value to be null when no values are present in the referencing objects. in kubebuilder marker, it can be done by adding `// +nullable` to the godoc, and the generators will be declaring the field as `nullable: true` in the openapi spec which allows the field to be `null` in yaml. by this way, we can discrinimate the case where the interest field of type string is an empty string `""` or the field doesnt exists (e.g. b/c no parent fields present).

Reply:
We will have 3 scenarios
1. the fieldValue is not present
2. the value is empty
3. the value is nil

My original thought is scenario 1 indicate there is a mismatch on field, or the value type is not supported. scenario 2 will indicate just empty value.

Reply:
by "misamatch", do you mean the parent field of the interested fields is not present on the resource? remember that a non-exist field can also be logically a nil field (e.g. if the field is tagged w/ `omitempty` in the go code), i presume practically it's not feasible to discriminate case (1) and (3) from the perspective of the work agent.

Reply:
Do you mean when we want feedback of `.status.foo.bar` we should discriminate between

```yaml
status:
  foo:
    bar: nil
```
and

```yaml
status:
  foor: {}
```

My original thought is we do not need to discriminate them by just skipping the related result, since none of them returned a meaningful result. Any reason we should discriminate them?

Reply:
just skipping should be fine in the first stage